client/src/www/app/outline_server_repository/server.ts

```diff
@@ -30,58 +31,58 @@ import {ResourceFetcher} from '../resource_fetcher';
 // PLEASE DON'T use this class outside of this `outline_server_repository` folder!
 
 export class OutlineServer implements Server {
-  errorMessageId?: string;
+  public readonly type: ServerType;
   readonly tunnelConfigLocation: URL;
-  private _address: string;
+  private displayAddress: string;
   private readonly staticTunnelConfig?: TunnelConfigJson;
+  errorMessageId?: string;
 
   constructor(
     private vpnApi: VpnApi,
     readonly urlFetcher: ResourceFetcher,
     readonly id: string,
     public name: string,
     readonly accessKey: string,
-    readonly type: ServerType,
     localize: Localizer
   ) {
-    switch (this.type) {
-      case ServerType.DYNAMIC_CONNECTION:
-        this.tunnelConfigLocation = new URL(
-          accessKey.replace(/^ssconf:\/\//, 'https://')
-        );
-        this._address = '';
-
-        if (!name) {
-          this.name =
-            this.tunnelConfigLocation.port === '443'
-              ? this.tunnelConfigLocation.hostname
-              : net.joinHostPort(
-                  this.tunnelConfigLocation.hostname,
-                  this.tunnelConfigLocation.port
-                );
-        }
-        break;
-
-      case ServerType.STATIC_CONNECTION:
-      default:
-        this.staticTunnelConfig = staticKeyToTunnelConfig(accessKey);
-        this._address = getAddressFromTransportConfig(
-          this.staticTunnelConfig.transport
-        );
+    const serviceConfig = parseAccessKey(accessKey);
+    this.name = name ?? serviceConfig.name;
+
+    if (serviceConfig instanceof DynamicServiceConfig) {
+      this.type = ServerType.DYNAMIC_CONNECTION;
+      this.tunnelConfigLocation = serviceConfig.transportConfigLocation;
+      this.displayAddress = '';
+
+      if (!serviceConfig.name) {
```

Comment:
Will we override user provided name here?

```suggestion
      if (!this.name) {
```

Reply:
Good catch. Fixed.